lib/search.js

```diff
@@ -50,7 +50,8 @@ function search(root, expression, maxDepth) {
 
     visit(root, HEADING, function (child, index, parent) {
         var value = toString(child);
-        var id = slugs.slug(value);
+        var id = child.data && child.data.hProperties && child.data.hProperties.id;
```

Comment:
I think this could be:

```js
var value = (child.data && child.data.hProperties && child.data.hProperties.id) || toString(child)
var id = slugs.slug(value)
```

...what do you think?

Reply:
That was my first attempt... but `value` is used later for the header title.  If you do that all links in the TOC would use `child.data.hProperties.id` as the text.